public/components/trace_analytics/components/traces/span_detail_panel.tsx

```diff
@@ -15,18 +15,34 @@ import {
   EuiSmallButton,
   EuiSpacer,
 } from '@elastic/eui';
-import debounce from 'lodash/debounce';
-import isEmpty from 'lodash/isEmpty';
 import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import useObservable from 'react-use/lib/useObservable';
 import { HttpSetup } from '../../../../../../../src/core/public';
 import { TraceAnalyticsMode } from '../../../../../common/types/trace_analytics';
 import { coreRefs } from '../../../../framework/core_refs';
 import { Plt } from '../../../visualizations/plotly/plot';
-import { handleSpansGanttRequest } from '../../requests/traces_request_handler';
 import { PanelTitle } from '../common/helper_functions';
 import { SpanDetailFlyout } from './span_detail_flyout';
 import { SpanDetailTable, SpanDetailTableHierarchy } from './span_detail_table';
+import { hitsToSpanDetailData } from '../../requests/traces_request_handler';
+
+/*
+ * Parse an ISO timestamp with up to nanosecond precision.
+ * For example, "2025-01-28T03:12:37.293990144Z" will be converted
+ * to a number representing the total nanoseconds since the Unix epoch.
+ */
+export function parseIsoToNano(iso: string): number {
+  const match = iso.match(/^(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2})(?:\.(\d+))?Z$/);
+  if (!match) {
+    throw new Error(`Invalid ISO timestamp: ${iso}`);
+  }
+  // Parse the base part using Date.parse (which gives ms)
+  const baseMs = new Date(match[1] + 'Z').getTime();
+  // Get the fractional part (if any), pad to 9 digits for nanosecond precision
+  let fraction = match[2] || '0';
+  fraction = fraction.padEnd(9, '0'); // ensure it has 9 digits
+  return baseMs * 1e6 + Number(fraction);
+}
```

Comment:
May be move this to helper function?

Reply:
Moved too public/components/trace_analytics/components/common/helper_functions.tsx